nomos-services/sdp/src/backends/sdpledger.rs

```diff
@@ -0,0 +1,61 @@
+use std::fmt::Debug;
+
+use nomos_sdp_core::{
+    ledger::{
+        DeclarationsRepository, RewardsRequestSender, SdpLedger, SdpLedgerError,
+        ServicesRepository, StakesVerifier,
+    },
+    BlockNumber, SdpMessage,
+};
+
+use super::SdpBackend;
+
+#[async_trait::async_trait]
+impl<Declarations, Rewards, Services, Stakes, Proof, Metadata, ContractAddress> SdpBackend
+    for SdpLedger<Declarations, Rewards, Services, Stakes, Proof, Metadata, ContractAddress>
+where
+    Declarations: DeclarationsRepository + Send + Sync + Clone + 'static,
+    Rewards: RewardsRequestSender<Metadata = Metadata, ContractAddress = ContractAddress>
+        + Send
+        + Clone
+        + Sync
+        + 'static,
+    Services: ServicesRepository<ContractAddress = ContractAddress> + Send + Sync + Clone + 'static,
+    Stakes: StakesVerifier<Proof = Proof> + Send + Clone + Sync + 'static,
+    ContractAddress: Debug + Send + Sync + 'static,
+    Proof: Send + Sync + 'static,
+    Metadata: Send + Sync + 'static,
+    SdpLedgerError<ContractAddress>: Send + Sync,
+{
+    type BlockNumber = BlockNumber;
+    type Message = SdpMessage<Metadata, Proof>;
+    type Error = SdpLedgerError<ContractAddress>;
+    type Settings = (Declarations, Rewards, Services, Stakes);
+
+    fn new(settings: Self::Settings) -> Self {
+        let (declaration_repo, reward_request_sender, services_repo, stake_verifier) = settings;
```

Comment:
It appears that precious design where passed overwatch handle will be required here, instead of getting these from settings, we'll need to create adapters for RewardsService and StakesService, the DeclarationsRepository should be part of SdpService implementation and probably use StorageService. Maybe for this PR lets leave the Mock, and create the SdpLedgerBackend in another one, where we use mock for rewards and staking, but write declarations and providers to real storage?

Reply:
Yeah, but the overwatch handle should still not be passed. All initialization should be done in the service layer. Backend should get just the adapter for whatever it needs. 
I agree we can go forward with a mock. But better mock adapters than keeping all this generics (which will spread) here.

Reply:
That's even better approach - @danielSanchezQ what do you think if we just remove the MockSdpLedgerBackend, pass adapters to the actual SdpLedgerBackend, and mock only rewards and staking adapters.

Reply:
> That's even better approach - @danielSanchezQ what do you think if we just remove the MockSdpLedgerBackend, pass adapters to the actual SdpLedgerBackend, and mock only rewards and staking adapters.

Yes, I think this is the way to do it.

Reply:
Hopefully I did it the way you guys proposed. I did not implement any mocks for now, just left a dummy todo impl for the actual adapters to keep the PR more contained. And we can see which direction to go next, based on what we have. Hope it makes sense